mypy/messages.py

```diff
@@ -992,6 +997,28 @@ def pretty_or(args: List[str]) -> str:
     return ", ".join(quoted[:-1]) + ", or " + quoted[-1]
 
 
+def invariance_notes(notes: List[str], arg_type: Type, expected_type: Type) -> List[str]:
+    """Explain that the type is invariant and give notes for how to solve the issue."""
+    if isinstance(arg_type, Instance) and isinstance(expected_type, Instance):
+        if (arg_type.type.fullname() == 'builtins.list' and
+                expected_type.type.fullname() == 'builtins.list'):
+            invariant_type = 'List'
+            covariant_suggestion = 'Consider using "Sequence" instead, which is covariant'
+        elif (arg_type.type.fullname() == 'builtins.dict' and
+              expected_type.type.fullname() == 'builtins.dict' and
+              arg_type.args[0].type == expected_type.args[0].type and
+              arg_type.args[1].type != expected_type.args[1].type):
+            invariant_type = 'Dict'
+            covariant_suggestion = ('Consider using "Mapping" instead, '
+                                    'which is covariant in the value')
```

Comment:
I am also not convinced that this is clear to a novice user (which I expect is the intended audience for this type of note), but I can't think of a better concise message. I guess this assumes that the link is the most helpful part.